File: app/src/main/java/ru/nikshlykov/englishwordsapp/ui/groups/GroupItem.java
package ru.nikshlykov.englishwordsapp.ui.groups;

import java.util.ArrayList;
import java.util.List;

import ru.nikshlykov.englishwordsapp.db.group.Group;
import ru.nikshlykov.englishwordsapp.db.subgroup.Subgroup;

public class GroupItem {
    private Group group;
    private ArrayList<Subgroup> subgroups;

    public GroupItem(Group group, ArrayList<Subgroup> subgroups) {
        this.group = group;
        this.subgroups = subgroups;
    }

    public void setGroup(Group group) {
        this.group = group;
    }

    public void setSubgroups(ArrayList<Subgroup> subgroups) {
        this.subgroups = subgroups;
    }

    public Group getGroup() {
        return group;
    }

    public ArrayList<Subgroup> getSubgroups() {
        return subgroups;
    }
}

File: feature-groups/src/main/java/ru/nikshlykov/feature_groups_and_words/domain/models/GroupItem.kt
package ru.nikshlykov.feature_groups_and_words.domain.models

import ru.nikshlykov.data.database.models.Group
import ru.nikshlykov.data.database.models.Subgroup

internal class GroupItem(var group: Group, var subgroups: ArrayList<Subgroup>) {
    override fun equals(obj: Any?): Boolean {
        if (this === obj) {
            return true
        }
        if (obj == null || this.javaClass != obj.javaClass) {
            return false
        }
        val comparedGroupItem = obj as GroupItem
        return if (group == comparedGroupItem.group) {
            val comparedSubgroups = comparedGroupItem.subgroups
            if (subgroups.size == comparedSubgroups.size) {
                for (i in subgroups.indices) {
                    if (subgroups[i] != comparedSubgroups[i]) {
                        return false
                    }
                }
                true
            } else {
                false
            }
        } else {
            false
        }
    }
}